src/extension/tools/vscode-node/toolsService.ts

```diff
@@ -113,5 +117,46 @@ export class ToolsService extends BaseToolsService {
 
 			return false;
 		});
+
+		// Extract semantic query from the request if available
+		let semanticQuery: string | undefined;
+		if (request.prompt) {
+			// Use the user's prompt as the semantic query
+			semanticQuery = request.prompt;
```

Comment:
I'm not sure the prompt has enough knowledge to do this off the bat. Imo the better data source for this is after the first LM turn where it'll usually plan out what it's going to do. It does mean we'd throw out the first cache breakpoint though, cc @roblourens

Reply:
Also misses the conversation and open files as context; which might carry a lot of meaning.